api/proto/teleport/legacy/types/types.proto

```diff
@@ -4008,8 +4006,9 @@ message TrustedClusterSpecV2 {
   // Setting Enabled to false has a side effect of deleting the user and host certificate
   // authority (CA).
   bool Enabled = 1 [(gogoproto.jsontag) = "enabled"];
-  // Roles is a list of roles that users will be assuming when connecting to this cluster.
-  repeated string Roles = 2 [(gogoproto.jsontag) = "roles,omitempty"];
+
+  reserved 2; // repeated string Roles - deprecated in favor of RoleMap
+
```

Comment:
Maybe? There's no way to reserve a `gogoproto.jsontag` tho, which is what we actually care about. 😢 

```suggestion
  reserved 2; // repeated string Roles - deprecated in favor of RoleMap
  reserved "Roles";
```